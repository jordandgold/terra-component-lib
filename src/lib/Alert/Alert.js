import React from "react";

import "./Alert.scss";

const Alert = props => {
  const { onClick, className, text } = props;

  return (
    <div
        onClick={props.onClick}
        className={`ter-alert ${className}`}
    >
        {text}
    </div>
  );
};

export default Alert;
