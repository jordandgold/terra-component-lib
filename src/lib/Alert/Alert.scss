@import '../global-styles/terra.scss';

@mixin alert-base() {
    padding: $alert-padding;
    display: block;
    margin-bottom: spacing('sm');
    position: relative;
    border-radius: $alert-border-radius;
    &__close {
        position: absolute;
        top: 1rem;
        right: 1rem;
        opacity: 0.4;
        padding: 0;
        margin: 0;
        background: transparent;
        border: 0;
        width: 20px;
        height: 20px;
        &:after {
            @include icon-base();
			content: '';
			position: absolute;
			top: 50%;
			right: 0px;
			transform: rotate(0deg) translateY(-50%);
			transition: transform 200ms ease-in-out;
			background-position: -316px -316px;
            height: 20px;
            width: 20px;
		}
        &:hover {
            opacity: 0.6;
            cursor: pointer;
        }
    }
}

@mixin alert-variant($color, $bg: transparent) {
    background-color: transparentize($color, .9);
    border-top: 1px solid darken($color, 10%);
    border-right: 1px solid darken($color, 10%);
    border-bottom: 3px solid darken($color, 10%);
    border-left: 1px solid darken($color, 10%);
    .ter-alert__icon {
        > i {
            &:before {
                color: darken($color, 20%);
            }
        }
    }
    .ter-alert__close {
        i {
            color: darken($color, 15%);
        }
    }
    hr {
        opacity: 0.4;
        background-color: darken($color, 10%);
    }
    a {
        color: darken($color, 20%);
        text-decoration: underline;
        &:hover {
            color: darken($color, 10%);
            text-decoration: underline;
        }
    }
}

.ter-alert {
    @include alert-base();
}

@each $color, $value in $emotive-colors {
  .ter-alert--#{$color} {
    @include alert-variant($value);
  }
}
