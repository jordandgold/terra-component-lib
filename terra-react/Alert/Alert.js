import React from "react";
import Icon from "../Icon/Icon";
import "./Alert.scss";

var Alert = function Alert(props) {
  var _onClick = props.onClick,
      type = props.type,
      text = props.text,
      name = props.name;
  return React.createElement("div", {
    className: "ter-alert ter-alert--".concat(type || "default")
  }, React.createElement("button", {
    className: "ter-alert__close",
    onClick: function onClick() {
      return _onClick(name);
    },
    "aria-label": "Close Alert"
  }, React.createElement(Icon, {
    type: "enclosed-x-dark-16px",
    size: "16px"
  })), text);
};

export default Alert;