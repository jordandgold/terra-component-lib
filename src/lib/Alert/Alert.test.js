import React from "react";
import { shallow } from "enzyme";
import Alert from "./Alert";

describe("Alert", () => {
  let wrapper;
  let mockOnClick;

  beforeEach(() => {
    mockOnClick = jest.fn();
    wrapper = shallow(
      <Alert
        text="This is an alert."
        type="ter-alert--default"
        onClick={mockOnClick}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call onClick when clicked", () => {
    wrapper.find(".ter-alert").simulate("click");
    expect(mockOnClick).toHaveBeenCalled();
  });
});
