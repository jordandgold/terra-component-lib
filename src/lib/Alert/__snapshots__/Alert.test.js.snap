// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should match snapshot without a type 1`] = `
<div
  className="ter-alert ter-alert--default"
  onClick={[MockFunction]}
>
  <button
    aria-label="Close Alert"
    className="ter-alert__close"
  >
    <Icon
      size="16px"
      type="enclosed-x-dark-16px"
    />
  </button>
  This is an alert.
</div>
`;

exports[`Alert should match the snapshot 1`] = `
<div
  className="ter-alert ter-alert--ter-alert--default"
  onClick={[MockFunction]}
>
  <button
    aria-label="Close Alert"
    className="ter-alert__close"
  >
    <Icon
      size="16px"
      type="enclosed-x-dark-16px"
    />
  </button>
  This is an alert.
</div>
`;
